enclave_manager/avalon_enclave_manager/singleton/singleton_enclave_manager.py

```diff
@@ -170,22 +175,30 @@ def _process_work_order_by_id(self, wo_id):
         self._kv_helper.remove("wo-scheduled", wo_id)
 
         logger.info("Validating JSON workorder request %s", wo_id)
-        validation_status = EnclaveManager.validate_request(wo_json_req)
+        validation_status, json_obj = EnclaveManager.validate_request(
+            wo_json_req)
 
         if not validation_status:
             logger.error(
                 "JSON validation for Workorder %s failed; " +
                 "handling Failure scenarios", wo_id)
-            wo_response = dict()
-            wo_response["Response"] = dict()
-            wo_response["Response"]["Status"] = WorkOrderStatus.FAILED
-            wo_response["Response"]["Message"] = \
-                "Workorder JSON request is invalid"
-            self._kv_helper.set("wo-responses", wo_id, json.dumps(wo_response))
-            self._kv_helper.set("wo-processed", wo_id,
-                                WorkOrderStatus.FAILED.name)
-            self._kv_helper.remove("wo-processing", wo_id)
-            return
+            self._handle_wo_process_failure(
+                "Workorder JSON request is invalid", wo_id)
+            return None
+
+        # Check if wo can be processed by this worker
+        if "params" in json_obj and "workloadId" in json_obj["params"]:
+            workload_id_in_req = json_obj["params"]["workloadId"]
+            if workload_id_in_req not in self._workloads:
```

Comment:
Can we add a function `can_worker_support_work_order(wo_request)`. This function can be reused by both WPE and Singleton. Could be kept in base enclave manager class